src/common/lib/client/realtime.ts

```diff
@@ -55,18 +54,39 @@ class Realtime extends Rest {
 class Channels extends EventEmitter {
   realtime: Realtime;
   all: Record<string, RealtimeChannel>;
-  inProgress: Record<string, RealtimeChannel>;
+  // Channel serials from the recovery key to be set when the channel is
+  // explicitly attached.
+  recoverySerials: { [name: string]: string };
 
   constructor(realtime: Realtime) {
     super();
     this.realtime = realtime;
     this.all = Object.create(null);
-    this.inProgress = Object.create(null);
+    this.recoverySerials = Object.create(null);
     realtime.connection.connectionManager.on('transport.active', () => {
       this.onTransportActive();
     });
   }
 
+  channelSerials(): { [ name: string ]: string } {
+    let serials: { [ name: string ]: string } = {};
+    for (const [name, channel] of Object.entries(this.all)) {
+      if (channel.channelSerial) {
+        serials[name] = channel.channelSerial;
+      }
+    }
+    return serials;
+  }
+
+  setRecoveryChannelSerials(channelSerials: { [ name: string ]: string }) {
+    for (const [name, serial] of Object.entries(channelSerials)) {
+      if (name in this.all) {
+        this.all[name].channelSerial = serial;
+      }
+      this.recoverySerials[name] = serial;
```

Comment:
We don't need `recoverySerials` instead the simplified code can be 
```
    for (const [name, serial] of Object.entries(channelSerials)) {
        this.get(name, null).channelSerial = serial;
      }
```

Reply:
`recoverySerials` var is introducing extra complexity and reduced code readability

Reply:
updated thanks!